sql/catalyst/src/main/scala/org/apache/spark/sql/internal/SQLConf.scala

```diff
@@ -812,6 +812,17 @@ object SQLConf {
     .intConf
     .createWithDefault(65535)
 
+  val CODEGEN_METHOD_SPLIT_THRESHOLD = buildConf("spark.sql.codegen.methodSplitThreshold")
+    .internal()
+    .doc("The threshold of source code length without comment of a single Java function by " +
+      "codegen to be split. When the generated Java function source code exceeds this threshold" +
+      ", it will be split into multiple small functions. We cannot know how many bytecode will " +
+      "be generated, so use the code length as metric. When running on HotSpot, a function's " +
+      "bytecode should not go beyond 8KB, otherwise it will not be JITted; it also should not " +
+      "be too small, otherwise there will be many function calls.")
+    .intConf
+    .createWithDefault(1024)
```

Comment:
let's add a check value to make sure the value is positive. We can figure out a lower and upper bound later.